pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -473,9 +472,8 @@ def save_function(self, value: Optional[Callable]) -> None:
         )
         self._save_function = value
 
-    @rank_zero_only
-    def _del_model(self, filepath: str) -> None:
-        if self._fs.exists(filepath):
+    def _del_model(self, trainer: 'pl.Trainer', filepath: str) -> None:
+        if self._fs.exists(filepath) and trainer.should_rank_save_checkpoint:
```

Comment:
```suggestion
        if trainer.should_rank_save_checkpoint and self._fs.exists(filepath):
```
in case self._fs is a remote filesystem (s3/aws/gcs), this could save a network call